selfdrive/car/gm/carstate.py

```diff
@@ -28,7 +28,11 @@ def update(self, pt_cp, loopback_cp):
     ret.vEgoRaw = mean([ret.wheelSpeeds.fl, ret.wheelSpeeds.fr, ret.wheelSpeeds.rl, ret.wheelSpeeds.rr])
     ret.vEgo, ret.aEgo = self.update_speed_kf(ret.vEgoRaw)
     ret.standstill = ret.vEgoRaw < 0.01
-    ret.gearShifter = self.parse_gear_shifter(self.shifter_values.get(pt_cp.vl["ECMPRDNL"]["PRNDL"], None))
+    if pt_cp.vl["ECMPRDNL2"]["ManualMode"] == 1:
+      ret.gearShifter = self.parse_gear_shifter("T")
+      ret.manumaticGear = pt_cp.vl["ECMPRDNL2"]["PRNDL2"]
```

Comment:
do we really need to add this? it's not used for anything

Reply:
I'm envisioning OP having the ability to adjust the accel/decel behavior as well as the max speed based on the selected manumatic gear. Similar to how we specify a min engage and set speed, while in manumatic mode we might might want to impose appropriate maximums.


The max gear selection changes the acceleration curve.
If it is set to 1 and you floor it, the car will take off fast but then probably top out at 25mph (I would expect the max speed for each gear is provided in the manual)

Whereas if it is 12 (the max on sback's Suburban) there is little difference from standard drive mode.

It sounds like manumatic might also provide engine braking.

Reply:
That would be a good fork feature, but let keep this to only required minimum changes for now. Sounds like we want the gear shifter change, as we disallow similar gears in other brands (low, sport, B on Toyota).

Reply:
Sure thing! I will close the cereal PR and update this one

Reply:
Done!